R/UK.R

```diff
@@ -123,7 +134,8 @@ UK <- R6::R6Class("UK", # rename to country name
           release_date = self$release_date
         )
       }
-      # get NHS data if requested
+
+      # merge NHS data if requested
       if (self$nhsregions) {
         self$add_nhs_regions()
```

Comment:
I think I would prefer this to do the join in this function vs internally. Having to many layers all referencing the same objects via `self` is pretty confusing.

Reply:
(in the absence of some docs explaining to me that its actually easy I find it confusing at least)

Reply:
so my thought would be that you pass in `self$data$clean` to `add_nhs_regions` operate on it (i.e joining in the NHS data) and then erasing to `self$data$clean` vs doing it all in function.

Reply:
So have the function take `self$data$cean` as an argument, add the nhs regions to it then return this and overwrite `self$data$clean`? 

e.g.

```
add_nhs_regions(clean_data) <- function(){
    clean_data >%> 'work on data set...'
    ...
    return(clean_data)
}

self$data$clean <- self$add_nhs_regions(self$data$clean)
```

Reply:
yes that is what I was thinking as that sort of structure might make understanding what is happening a bit easier... maybe even 

```
self$data$clean <- self$add_nhs_regions(self$data$clean, self$data$nhs_raw)
```

if getting extreme about it.

Reply:
I will fix here now, but apply changes in a seperate PR for #201

Reply:
* Apply changes to other functions I mean